companion/lib/ImportExport/Controller.ts

```diff
@@ -488,22 +488,23 @@ export class ImportExportController {
 
 						if (data.type !== 'full') throw new Error('Invalid import object')
 
-						const resetArg: ClientResetSelection | null =
-							fullReset || !config ? null : { ...config, connections: true, userconfig: false }
-						if (resetArg && !fullReset) {
-							// Dont ever reset connections during partial import
-							resetArg.connections = false
-
-							// If a partial import, don't reset disabled items
-							if (!data.pages) resetArg.buttons = false
-							if (!data.custom_variables) resetArg.customVariables = false
-							if (!data.expressionVariables) resetArg.expressionVariables = false
-							if (!data.surfaces) resetArg.surfaces = false
-							if (!data.triggers) resetArg.triggers = false
+						// `config` tells what to load. Ensure that config is false for missing sections:
+						// note: this is failsafe is not strictly necessary, since Full.tsx does it right, now.
+						// note that config doesn't have a entries for 'connections' or 'userconfig'...
+						for (const key in config) {
+							let dataKey = key.replace(/^customVariables$/, 'custom_variables')
+							dataKey = dataKey.replace(/^buttons$/, 'pages')
```

Comment:
I'm not the biggest fan of this, because I know I will forget to update it when renaming things inside the exports in the future.
Equally, that was a problem with what I had before and being easy to forget to add new things, sooo 🤷

Reply:
I completely agree with your sentiment, but don't see any "correct" way to fix it without breaking either the export file or the internal database. The current code does make it more compact, and a bit easier to see the "problem" (i.e. a little harder to forget...).

That said, the problem is not related to this PR, and has existed since at least the mid v3's, if memory serves, so perhaps you could post an Issue, and either suggest or we can discuss a solution outside this PR?  (I'm happy for you to assign it to me!)